src/action.rs

```diff
@@ -75,17 +71,55 @@ pub enum Action {
     PeerTryWrite(PeerTryWriteAction),
     PeerTryRead(PeerTryReadAction),
 
-    PeerHandshakingInit(PeerHandshakingInitAction),
-
-    PeerConnectionMessageWriteInit(PeerConnectionMessageWriteInitAction),
-    PeerConnectionMessagePartWritten(PeerConnectionMessagePartWrittenAction),
-    PeerConnectionMessageWriteError(PeerConnectionMessageWriteErrorAction),
-    PeerConnectionMessageWriteSuccess(PeerConnectionMessageWriteSuccessAction),
+    // chunk read
+    PeerChunkReadInit(PeerChunkReadInitAction),
+    PeerChunkReadPart(PeerChunkReadPartAction),
+    PeerChunkReadDecrypt(PeerChunkReadDecryptAction),
+    PeerChunkReadReady(PeerChunkReadReadyAction),
+    PeerChunkReadError(PeerChunkReadErrorAction),
+
+    // chunk write
+    PeerChunkWriteSetContent(PeerChunkWriteSetContentAction),
+    PeerChunkWriteEncryptContent(PeerChunkWriteEncryptContentAction),
+    PeerChunkWriteCreateChunk(PeerChunkWriteCreateChunkAction),
+    PeerChunkWritePart(PeerChunkWritePartAction),
+    PeerChunkWriteReady(PeerChunkWriteReadyAction),
+    PeerChunkWriteError(PeerChunkWriteErrorAction),
+
+    // binary message read
+    PeerBinaryMessageReadInit(PeerBinaryMessageReadInitAction),
+    PeerBinaryMessageReadChunkReady(PeerBinaryMessageReadChunkReadyAction),
+    PeerBinaryMessageReadSizeReady(PeerBinaryMessageReadSizeReadyAction),
+    PeerBinaryMessageReadReady(PeerBinaryMessageReadReadyAction),
+    PeerBinaryMessageReadError(PeerBinaryMessageReadErrorAction),
+
+    // binary message write
+    PeerBinaryMessageWriteSetContent(PeerBinaryMessageWriteSetContentAction),
+    PeerBinaryMessageWriteNextChunk(PeerBinaryMessageWriteNextChunkAction),
+    PeerBinaryMessageWriteReady(PeerBinaryMessageWriteReadyAction),
+    PeerBinaryMessageWriteError(PeerBinaryMessageWriteErrorAction),
 
-    PeerConnectionMessageReadInit(PeerConnectionMessageReadInitAction),
-    PeerConnectionMessagePartRead(PeerConnectionMessagePartReadAction),
-    PeerConnectionMessageReadError(PeerConnectionMessageReadErrorAction),
-    PeerConnectionMessageReadSuccess(PeerConnectionMessageReadSuccessAction),
+    PeerHandshakingInit(PeerHandshakingInitAction),
+    PeerHandshakingConnectionMessageInit(PeerHandshakingConnectionMessageInitAction),
+    PeerHandshakingConnectionEncode(PeerHandshakingConnectionEncodeAction),
```

Comment:
Please add `Message` to all of the other actions: `PeerHandshakingConnectionMessageEncode`, `PeerHandshakingConnectionMessageWrite`, etc... same for meta and ack messages.